src/library_async.js

```diff
@@ -247,7 +250,14 @@ mergeInto(LibraryManager.library, {
           if (typeof Browser !== 'undefined' && Browser.mainLoop.func) {
             Browser.mainLoop.resume();
           }
-          var asyncWasmReturnValue = Asyncify.doRewind(Asyncify.currData);
+          var asyncWasmReturnValue, isError = false;
+          try {
+            asyncWasmReturnValue = Asyncify.doRewind(Asyncify.currData);
+          } catch (err) {
+            asyncWasmReturnValue = err;
+            isError = true;
+          }
+          var handled = false;
```

Comment:
Can you add a comment for `handled`? Reading the code I'm not sure what that means. (It seems to be `!Asyncify.currData && Asyncify.asyncPromiseHandlers`, but I don't know how to interpret that...)

Reply:
Rather `!Asyncify.currData && !Asyncify.asyncPromiseHandlers` - that is, if there was no handler, then rethrow as regular exception.

Reply:
Added comments.